src/Lean/Elab/Term.lean

```diff
@@ -119,7 +118,8 @@ structure State where
   levelNames        : List Name       := []
   syntheticMVars    : MVarIdMap SyntheticMVarDecl := {}
   pendingMVars      : List MVarId := {}
-  mvarErrorInfos    : MVarIdMap MVarErrorInfo := {}
+  mvarErrorInfos    : List MVarErrorInfo := []
```

Comment:
Adding motivation for the modification is also helpful. That is, comments such as "We used to implement ... using ..., but it had problems with ... because of .... The new approach fixes the issue using ..." greatly improve maintainability.
We found that adding this kind of comment greatly improves maintainability.

Reply:
Ok, I've added some documentation motivating the new fields.